src/mlpack/methods/naive_bayes/naive_bayes_classifier_impl.hpp

```diff
@@ -195,18 +194,25 @@ void NaiveBayesClassifier<MatType>::Classify(const MatType& data,
 
     // Calculate probability as sum of logarithm to decrease floating point
     // errors.
-    testProbs.col(i) += (data.n_rows / -2.0 * log(2 * M_PI) - 0.5 *
+    testProbs.col(i) += (data.n_rows / - 2.0 * log(2 * M_PI) - 0.5 *
         log(arma::det(arma::diagmat(variances.col(i)))) + exponents);
   }
 
+  arma::vec log_prob_x(data.n_cols); // log(Prob(X))
+  for (size_t j = 0; j < data.n_cols; ++j)
+    log_prob_x(j) = log(arma::accu(exp(testProbs.row(j))));
+
+  testProbs -= arma::repmat(log_prob_x, 1, testProbs.n_cols);
```

Comment:
This calculates total probability of features. Since we use log probability, we need to subtract it. I will add some comments to clarify this.